### ISSUE TITLE
Destroy method

### ISSUE THREAD

**Issue by scorpionfire** [jaredreich/pell#112]
Can be an idea add a **destroy** method. I need this method because I use this library in a SPA website.

The method should remove the events and remove the HTML injected.

Thanks

**Comment by jaredreich** [comment]
Thanks for the suggestion. Out of curiosity, when would you run this `destroy` method?

**Comment by scorpionfire** [comment]
I have various cases. 

For example, when I open a page, I use Knockout to render it. after the render I init Pell. 
Then, when I change page, before the new ajax call, I clar all events and delete elements of the previous content. In this moment I need to clear all events of Pell (and remove the HTML it ineject in the page).

Thanks 🦊

**Comment by jaredreich** [comment]
Interesting. This sounds like stuff that comes out-of-the-box with frameworks like Vue and React. Old elements (and all events tied to those elements) are destroyed upon rendering a different route/page. Hmmmm...

**Comment by scorpionfire** [comment]
Yes, like Angular, Ract, etc ... but we have our own framework, so I need to conteplate all external plugin to use them and the destroy them.

Thanks

**Comment by jaredreich** [comment]
So I don't think this should be the responsibility of pell. If you delete the HTML that pell is contained within, all the related events will subsequently be destroyed, since they are attached to those elements.

See https://developer.mozilla.org/en-US/docs/Web/API/Node/removeChild

**Comment by jaredreich** [comment]
It appears I may be wrong according to https://stackoverflow.com/questions/14040386/should-i-remove-event-handlers-from-element-before-removechild

Re-opening.

**Comment by jaredreich** [comment]
More relevant links:
https://stackoverflow.com/questions/12528049/if-a-dom-element-is-removed-are-its-listeners-also-removed-from-memory
https://stackoverflow.com/questions/4386300/javascript-dom-how-to-remove-all-events-of-a-dom-object

**Comment by stephankaag** [comment]
Is there an ETA for this issue?

**Comment by mgiraldo** [comment]
any idea how to address this issue? i have an html text block that when the user clicks becomes pell-editable (vue 3). i need to be able to destroy the current editor when the user goes from "view mode" to "edit mode"
